src/controllers/controller.bar.js

```diff
@@ -328,18 +328,13 @@ module.exports = function(Chart) {
 			var me = this;
 			var chart = me.chart;
 			var rects = me.getMeta().data;
-			var dataset = me.getDataset();
 			var ilen = rects.length;
 			var i = 0;
-			var d;
 
 			helpers.canvas.clipArea(chart.ctx, chart.chartArea);
 
 			for (; i < ilen; ++i) {
-				d = dataset.data[i];
-				if (d !== null && d !== undefined && !isNaN(d)) {
```

Comment:
Looks like a regression, `null/undefined/NaN` data are currently used to skip element (@etimberg?) Should be the same logic as your other change: `d = scale.getRightValue(dataset.data[i])`.

I doubt that's the responsability of the elements to decide whether they needs to drawn itself. The controller seems a better candidate, each controller can have different behavior.